### PR TITLE
make_clickable(): Support URL with a closing parenthesis before a media/file extension

### PR DESCRIPTION
**Pull request by hellofromtonya**
## Tests: `make_clickable()` tests: Adds datasets for URLs with `()` within the path and with extensions.

A real world example image:
https://en.wikipedia.org/wiki/HMS_Ark_Royal_(91)#/media/File:HMS_Ark_Royal_(91)_just_after_launching.jpg

## `make_clickable()`: Support URL with a closing parenthesis before a media/file extension.

Patch https://core.trac.wordpress.org/attachment/ticket/62037/62037-2.patch

Co-authored-by: PARTHVATALIYA <parthvataliya@git.wordpress.org>

Trac ticket: https://core.trac.wordpress.org/ticket/62037

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/formatting.php [hunk 1]
```diff
@@ -2943,6 +2943,10 @@ function _make_url_clickable_cb( $matches ) {
 		$suffix = $matches[3];
 	}
 
+	if ( isset( $matches[4] ) && ! empty( $matches[4] ) ) {
```
Comment (hellofromtonya) [inline src/wp-includes/formatting.php:2946]:
```suggestion
	if ( ! empty( $matches[4] ) ) {
```

`empty()` also checks if the element is exists.

File: src/wp-includes/formatting.php [hunk 2]
```diff
@@ -3115,6 +3119,7 @@ function make_clickable( $text ) {
 					)*
 				)
 				(\)?)                                          # 3: Trailing closing parenthesis (for parenthesis balancing post processing).
+				(\\.\\w{2,6})?               # Allowing file extensions (e.g., .jpg, .png).
```
Comment (hellofromtonya) [inline src/wp-includes/formatting.php]: The inline comment will need alignment with the other inline comments, and likely needs to be labeled as `# 4:`

**Review by peterwilsoncc — COMMENTED** [review]
This looks good to me and tests well using the playground link.

I'll push up a change to the comment.

**Comment by peterwilsoncc** [comment]
Hmm, the tests started failing after merging in trunk and making the whitespace change to the comments. As this is a bug, I'll leave it for post beta 1.

**Comment by peterwilsoncc** [comment]
Failures are unrelated, committing.

**Comment by peterwilsoncc** [comment]
Merged [r59143](https://core.trac.wordpress.org/changeset/59143) / https://github.com/WordPress/wordpress-develop/commit/9b6c034e2fbb3bb876932cf3770a93791a7a22f3

**Pull request closed without merging**